hasher-matcher-actioner/hmalib/matchers/matchers_base.py

```diff
@@ -114,9 +150,34 @@ def filter_match_results(self, results: t.List[IndexMatch]) -> t.List[IndexMatch
             )
 
             if len(match.metadata["privacy_groups"]) != 0:
-                filtered_results.append(match)
+                filtered_results_based_on_active.append(match)
+
+        # Also check and see if there is a custom threshold filter
+        filtered_results_based_on_threshold = []
+        for match in filtered_results_based_on_active:
+            filtered_privacy_groups = []
+            for privacy_group in match.metadata.get("privacy_groups", []):
+                if threshold := get_optional_privacy_group_matcher_pdq_theshold(
+                    str(privacy_group)
+                ):
+                    if match.distance >= threshold:
+                        logger.debug(
+                            "Match was found at %s distance but exceeds custom theshold of %s for pg, %s",
+                            match.distance,
+                            threshold,
+                            privacy_group,
+                        )
+                    else:
+                        filtered_privacy_groups.append(privacy_group)
+                else:
+                    # no custom threshold found
+                    filtered_privacy_groups.append(privacy_group)
```

Comment:
If index was created when privacy_group's threshold was defined at say 40, and the matcher kicked in when the threshold was removed, does this mean the default threshold of 31 will be ignored?

Right now, we rebuild indexes frequently enough, but soon, that might not be the case. If so, we are introducing an implicit dependency. 

Might be totally wrong though. :)

Reply:
Not sure I understand your questions, if no custom threshold is found we do NOT filter so the default 31 would be respected and a match recorded. these changes don't care about index building; they are built independent of any pg's thresholds